specification/communication/data-plane/CallAutomation/stable/2024-04-15/communicationservicescallautomation.json

```diff
@@ -0,0 +1,3076 @@
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "CallLocatorKind",
+        "modelAsString": true
+      }
+    },
+    "CallLocator": {
+      "description": "The locator used for joining or taking action on a call.",
+      "type": "object",
+      "properties": {
+        "groupCallId": {
+          "description": "The group call id",
+          "type": "string"
+        },
+        "serverCallId": {
+          "description": "The server call id.",
+          "type": "string"
+        },
+        "kind": {
+          "$ref": "#/definitions/CallLocatorKind"
+        }
+      }
+    },
+    "ChannelAffinity": {
+      "description": "Channel affinity for a participant",
+      "required": [
+        "participant"
+      ],
+      "type": "object",
+      "properties": {
+        "channel": {
+          "format": "int32",
+          "description": "Channel number to which bitstream from a particular participant will be written.",
+          "maximum": 4,
+          "minimum": 0,
+          "type": "integer"
+        },
+        "participant": {
+          "$ref": "#/definitions/CommunicationIdentifierModel",
+          "description": "The identifier for the participant whose bitstream will be written to the channel \r\nrepresented by the channel number."
+        }
+      }
+    },
+    "RecordingStorageKind": {
+      "description": "Defines the kind of external storage",
+      "enum": [
+        "azureCommunicationServices",
+        "azureBlobStorage"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "RecordingStorageKind",
+        "modelAsString": true
+      }
+    },
+    "ExternalStorage": {
+      "required": [
+        "recordingStorageKind"
+      ],
+      "type": "object",
+      "properties": {
+        "recordingStorageKind": {
+          "$ref": "#/definitions/RecordingStorageKind"
+        },
+        "azureBlobStorage": {
+          "$ref": "#/definitions/AzureBlobStorage",
+          "description": "Defines the Azure Blob storage location where the recording will be stored"
+        }
+      }
+    },
+    "AzureBlobStorage": {
+      "description": "Used to specify Azure Blob container url to recording storage",
+      "required": [
+        "containerUri"
+      ],
+      "type": "object",
+      "properties": {
+        "containerUri": {
+          "description": "Url of a container or a location within a container",
+          "type": "string"
+        }
+      }
+    },
+    "RecordingState": {
+      "enum": [
+        "active",
+        "inactive"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "RecordingState",
+        "modelAsString": true
+      }
+    },
+    "RecordingKind": {
+      "enum": [
+        "azureCommunicationServices",
+        "teams",
+        "teamsCompliance",
+        "others"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "RecordingKind",
+        "modelAsString": true
+      }
+    },
```

Comment:
You can include "values" in the "x-ms-enum" with a description for each value, and these descriptions will then be shown in the generated REST API Docs. Descriptions for enum values will improve the developer experience for your service. See the [autorest documentation][] for the syntax of the "values" field in "x-ms-enum".

[autorest documentation]: https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-enum